cmd/thanos/rule.go

```diff
@@ -377,7 +390,8 @@ func runRule(
 		}
 		fanoutStore := storage.NewFanout(logger, agentDB, remoteStore)
 		appendable = fanoutStore
-		queryable = fanoutStore
+		// Use a separate queryable to restore the ALERTS firing states.
```

Comment:
Nit: Could we maybe just add small note to make clear why (because we cannot read from `remoteStore`, basically to avoid same confusion I had in https://github.com/thanos-io/thanos/pull/5230#discussion_r916782619).

Reply:
Done